src/NLog/Targets/TargetWithContext.cs

```diff
@@ -0,0 +1,797 @@
+        public bool IncludeEventProperties { get => _contextLayout.IncludeAllProperties; set => _contextLayout.IncludeAllProperties = value; }
+
+        /// <inheritdoc/>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeMdc { get => _contextLayout.IncludeMdc; set => _contextLayout.IncludeMdc = value; }
+
+        /// <inheritdoc/>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeNdc { get => _contextLayout.IncludeNdc; set => _contextLayout.IncludeNdc = value; }
+
+#if !SILVERLIGHT
+        /// <inheritdoc/>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeMdlc { get => _contextLayout.IncludeMdlc; set => _contextLayout.IncludeMdlc = value; }
+
+        /// <inheritdoc/>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeNdlc { get => _contextLayout.IncludeNdlc; set => _contextLayout.IncludeNdlc = value; }
+#endif
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to include contents of the <see cref="GlobalDiagnosticsContext"/> dictionary
+        /// </summary>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeGdc { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to include call site (class and method name) in the <see cref="LogEventInfo" />
+        /// </summary>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeCallSite { get => _contextLayout.IncludeCallSite; set => _contextLayout.IncludeCallSite = value; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to include source info (file name and line number) in the <see cref="LogEventInfo" />
+        /// </summary>
+        /// <docgen category='Layout Options' order='10' />
+        public bool IncludeCallSiteStackTrace { get => _contextLayout.IncludeCallSiteStackTrace; set => _contextLayout.IncludeCallSiteStackTrace = value; }
+
+        /// <summary>
+        /// Gets the array of custom attributes to be passed into the logevent context
+        /// </summary>
+        /// <docgen category='Layout Options' order='10' />
+        [ArrayParameter(typeof(TargetWithContextProperty), "contextproperty")]
+        virtual public IList<TargetWithContextProperty> ContextProperties { get; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        protected TargetWithContext()
+        {
+            _contextLayout = _contextLayout ?? new TargetWithContextLayout(this, base.Layout);
+            OptimizeBufferReuse = true;
+        }
+
+        /// <summary>
+        /// Check if logevent has properties (or context properties)
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <returns>True if properties should be included</returns>
+        protected bool HasCombinedProperties(LogEventInfo logEvent)
```

Comment:
proposal to rename (after renames below) -> ShouldIncludeProperties

I prefer removing the term "combined properties", if possible

Reply:
Done